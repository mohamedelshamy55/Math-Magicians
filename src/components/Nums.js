import React from 'react';
import PropTypes from 'prop-types';

const Nums = ({ clickHandler, buttonName }) => (
  <>
    <li data-testid="num-li" btnname={buttonName}><button type="button" onClick={clickHandler}>{buttonName}</button></li>
  </>
);

export default Nums;

Nums.propTypes = {
  buttonName: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};
