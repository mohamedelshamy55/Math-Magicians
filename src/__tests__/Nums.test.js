import React from 'react';
import { render } from '@testing-library/react/';
import '@testing-library/jest-dom/extend-expect';
import Nums from '../components/Nums';

const renderer = require('react-test-renderer');

it('renders the props of component correctly', () => {
  const clickHandler = () => true;
  const buttonName = '5';
  const { getByTestId } = render(<Nums
    buttonName={buttonName}
    clickHandler={clickHandler}
  />);
  expect(getByTestId('num-li')).toHaveTextContent('5');
});

it('matches snapshot 1 of Num component', () => {
  const clickHandler = () => true;
  const buttonName = '5';
  const tree = renderer.create(<Nums
    buttonName={buttonName}
    clickHandler={clickHandler}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
