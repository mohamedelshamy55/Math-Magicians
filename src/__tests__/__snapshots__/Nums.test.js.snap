// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 of Num component 1`] = `
<li
  btnname="5"
  data-testid="num-li"
>
  <button
    onClick={[Function]}
    type="button"
  >
    5
  </button>
</li>
`;
